pkg/apps/util/util.go

```diff
@@ -276,6 +345,76 @@ func CopyApiEnvVarToV1EnvVar(in []api.EnvVar) []v1.EnvVar {
 	return out
 }
 
+func CopyPodTemplateSpecToV1PodTemplateSpec(spec *api.PodTemplateSpec) *v1.PodTemplateSpec {
+	copied, err := api.Scheme.DeepCopy(spec)
+	if err != nil {
+		panic(err)
+	}
+	in := copied.(*api.PodTemplateSpec)
+	out := &v1.PodTemplateSpec{}
+	if err := v1.Convert_api_PodTemplateSpec_To_v1_PodTemplateSpec(in, out, nil); err != nil {
+		panic(err)
+	}
+	return out
+}
+
+// HasLatestPodTemplate checks if the provided replication controller has the latest
+// deployment config pod template. If not it will return false and the string diff.
+func HasLatestPodTemplate(dc *deployapi.DeploymentConfig, rc *v1.ReplicationController) (bool, string, error) {
+	configTemplate := CopyPodTemplateSpecToV1PodTemplateSpec(dc.Spec.Template)
+	configTemplate = filterOutTemplateAnnotationsAndLabels(configTemplate)
+	rcCopy, err := DeploymentDeepCopyV1(rc)
+	if err != nil {
+		return true, "", err
+	}
+	rcTemplate := filterOutTemplateAnnotationsAndLabels(rcCopy.Spec.Template)
+	if reflect.DeepEqual(configTemplate, rcTemplate) {
+		return true, "", nil
+	}
+	return false, diff.ObjectReflectDiff(configTemplate, rcTemplate), nil
+}
+
+// HasUpdatedImages indicates if the deployment configuration images were updated.
+func HasUpdatedImages(dc *deployapi.DeploymentConfig, rc *v1.ReplicationController) (bool, []string) {
+	updatedImages := []string{}
+	rcImages := sets.NewString()
+	for _, c := range rc.Spec.Template.Spec.Containers {
+		rcImages.Insert(c.Image)
+	}
+	for _, c := range dc.Spec.Template.Spec.Containers {
+		if !rcImages.Has(c.Image) {
+			updatedImages = append(updatedImages, c.Image)
+		}
+	}
+	if len(updatedImages) == 0 {
```

Comment:
@mfojtik maybe this function could return just the array as the bool property is already contained in its length?

Reply:
i would rather prefer to be explicit as the function has `Has*` prefix, I would expect boolean.

Reply:
I though rename was self suggesting for that case